pilot/pkg/config/clusterregistry/multicluster.go

```diff
@@ -39,7 +40,8 @@ type Multicluster struct {
 	ResyncPeriod      time.Duration
 	serviceController *aggregate.Controller
 	ClearCache        func()
-	rkc               map[string]*kubeController
+	sync.Mutex
+	rkc map[string]*kubeController
```

Comment:
@john-a-joyce I really donot like this name? How about `controllers` `controllerMap` or some other meaningful?

Reply:
@hzxuzhonghu - I fixed this up again to avoid argument but IMO - its bad practice to be commenting and asking for changes to areas of the code that weren't changed initially.  It causes the code base to be in a constant state of churn which is not good.  Again IMO.   Look at the amount of churn we have had even on this simple change.